File: MAP(Advanced Programming Methods)/countdownlatch/src/model/stmt/IfStatement.java
package model.stmt;

import model.PrgState;
import model.expr.Expression;

public class IfStatement implements Statement {

    private Expression expr;
    private Statement thenstmt;
    private Statement elsestmt;

    public IfStatement(Expression e,Statement then,Statement el)
    {
        this.expr = e;
        this.thenstmt = then;
        this.elsestmt = el;
    }

    @Override
    public PrgState execute(PrgState p) {
        int result = this.expr.evaluate(p.getSymbolTable(),p.getHeap());
        if(result !=0)
            p.getExeStack().push(thenstmt);
        else
            p.getExeStack().push(elsestmt);

        return null;
    }

    public String toString(){
        return "if("+ expr+") then(" +thenstmt+")ELSE("+elsestmt+")";

    }
}

File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/Statements/IfStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Model;
using LabAssig7.Model.ADT;
using LabAssig7.Model.Expressions;
namespace LabAssig7.Model.Statements
{
    public class IfStatement:Statement
    {
        private Expression expr;
        private Statement elsestmt,thenstmt;

        public IfStatement(Expression e,Statement stmt1, Statement stmt2)
        {
            this.expr = e;
            this.elsestmt = stmt1;
            this.thenstmt = stmt2;
        }

        public PrgState Execute(PrgState p)
        {
            ISymbolTable<string, int> dict = p.Dict;
            int val = this.expr.Eval(dict);
            if(val !=0)
            {
                p.Stack.Push(thenstmt);
            }
            else
            {
                p.Stack.Push(elsestmt);
            }
            return p;
        }

        public override string ToString()
        {
            StringBuilder buff = new StringBuilder();
            buff.Append("if(");
            buff.Append(this.expr);
            buff.Append("{");
            buff.Append(thenstmt);
            buff.Append("}else{");
            buff.Append(this.elsestmt);
            buff.Append("}\n");
            return buff.ToString();
        }
    }
}
